analysis/Proposed_indicators.md

```diff
@@ -1,6 +1,83 @@
+    stats = ox.basic_stats(subgraph, area=area_sqm, clean_intersects=True, tolerance=intersection_tolerance,
+                circuity_dist='euclidean')
+    return({'graph': subgraph,
+             'network_gdfs':su,
+             'buffer':uu,
+             'area_sqkm':area_sqkm,
+             'stats':stats,
+             'origin_node_snap_dist': orig_node[1]})
+     
+# Define some point locations over which we'll iterate
+point_locations = [(-37.808156, 144.964942),(-37.764727, 144.964057)]
+# initialise an 
+nh_estimates = []
+
+for point in point_locations:
+  start = time.time()
+  task = "Calculate local neighbourhood measures for points"
+  nh_estimates.append(analyse_local_nh(G, point))
+  log('Completed task "{}" in {:,.2f} seconds'.format(task,time.time() - start)) 
+  
+# Get some stats
+## area in sqkm
+[x['area_sqkm'] for x in nh_estimates]
+
+## intersection density per sqkm
+[x['stats']['intersection_density_km'] for x in nh_estimates]
+
+## clean intersection density per sqkm
+[x['stats']['clean_intersection_density_km'] for x in nh_estimates]
+```
+So, that makes use of the osmnx stats functionality for calculating street connectivity.  
+
+The use of the function might be overkill; assuming we have a dataframe of sample points, if you coded it right there might even be a vectorised solution to carry out the operations, like extracting the 50m buffered unary union of 1600m network linestrings for a sample point as a geometry field and then use intersection analysis for each sample point's network buffer polygon with other pre-prepared resources:
+
+- use population raster (eg worldpop, or other) to derive a scaled-population density measure to serve as a proxy measure for dwelling density (e.g. take sum of population cells intersecting sausage buffer and divide by some indicative number for population per dwelling (e.g. 4) then standardise that result by dividing by the area of the sausage buffer in Hectares  --- so a result a-bit like dwellings per hectare (although technically, it isn't that.... i'm not saying this is valid, but it could be done and might serve for this purpose)
+
+- count clean network intersections generated by osmnx which intersect a sample point's sausage buffer and divide by area in km2 for street connectivity per square kilometre
+
+- if we pre-extract a series of destination POIs grouped into something like (pending discussion on how to translate this globally) 'supermarkets', 'convenience', 'public transport stops' (the latter from GTFS analysis - possibly restricted to those with 'frequent' service) then calculate a score for each sausage buffer based on presence of at least one from each category.
+
+Then, once you have a dataframe with rows per sample points with each associated with the above measures we could calculate their respective z-scores as per [this post](https://stackoverflow.com/questions/24761998/pandas-compute-z-score-for-all-columns) you could do something like
```

Comment:
Hi, Carl, I may have a misunderstanding here. Could you explain to me what you mean by getting the `dataframe measure with rows per sample points`? Do you mean we will have the walkability proxies (street connectivity, dwelling density, daily living scores) calculated for each sample point, and then taking the z scores for each column measure. As for the final walkability indicator, we will take the average of z-scores for each column and average again to get the final walkability score?

Reply:
That's similar but different to what i meant --- actually, it might resolve to the same thing!  

Yes, calculate street connectivity (sc), dwelling density (dd) and daily living (dl) for each sample point.

The next step is actually a bit tricky, and i've recently been considering it in a bit more detail:  rather than just taking the z-score of each measure for each sample point, it would be a good idea to first evaluate: are all sample points valid?  There may be connectivity or data quality related reason why some sample points may not be valid, despite having been generated on the network: e.g. they may be located on a network island (not an issue if retain_all = False) or have some other data issue meaning we have doubt about the validity of that point's measures (e.g. the points may be in an area of the city which doesn't have OSM destination coverage --- most likely in non-urban portions of the city; this is why we use an official data source in Australia to restrict to 'urban' regions.  I'm not sure how to generalise this... but worth thinking about).

So, before calculating z-scores I think it might be important to restrict the sample points to the subset of sample points which we consider to actually be valid (ie. those with IDs not in some kind of exclusion list which has been populated programmatically, and optionally manually with points matching exclusion criteria like "no estimates for distance to closest" which shouldn't be possible under normal circumstances).  We should discuss what those exclusion/inclusion criteria might be.

Then, we calculate z-scores and sum up the z-scores of sc, dd and dl for each point.  That gives the walkability index for each point.  This can be averaged for small areas, or whole region, or possibly using a population weighted average.  That's my understanding of the process...

Reply:
That's a good point about the sample point selection. Geoff and I have discussed this too. A bigger question here may be the official data availability. I know the U.S. Census specifies block-level urban and rural population, so we can always use this to specify the urban portion/urbanized areas of the study region, but it may not work for every other city. Other solution may be the population-grid approach as you mentioned for a weighted sample. Anyway, I like your idea to have this step done after the proxy calculation for each sample point. Will think about this as we move forward!